tests/unit/test_utils.py

```diff
@@ -366,11 +366,11 @@ def test_rmtree_retries(tmpdir, monkeypatch):
     rmtree('foo')
 
 
-def test_rmtree_retries_for_3sec(tmpdir, monkeypatch):
+def test_rmtree_retries_for_12sec(tmpdir, monkeypatch):
```

Comment:
Does this test actually last 12 seconds?

Reply:
yes, sadly.
```
>tox -e py37 -- tests\unit\test_utils.py
. . .
py37 run-test: commands[0] | pytest --timeout 300 'tests\unit\test_utils.py'
============================= test session starts =============================
platform win32 -- Python 3.7.4, pytest-3.8.2, py-1.8.0, pluggy-0.13.0
. . .
tests\unit\test_utils.py .............................s................. [ 32%]
........................................................................ [ 82%]
..........................                                               [100%]
=========================== short test summary info ===========================
SKIP [1] tests\unit\test_utils.py:400: condition: sys.platform == 'win32'

=================== 144 passed, 1 skipped in 15.06 seconds ====================
_______________________________________________________________ summary _______________________________________________________________
  py37: commands succeeded
  congratulations :)
```

Reply:
Ouch. Can we refactor like this:

```
def rmtree_provider(**kwargs):
    @retry(**kwargs)
    def rmtree(...):
        # as before
        ...

   return rmtree
```

and just test that it works as expected with 1 or 2 seconds, then set `rmtree = rmtree_provider(stop_max_delay=12000, wait_fixed=500)` in `pip._internal.utils.misc`? Then we're free to adjust the time without impacting test duration.

If you want to make that as a separate PR it would help separate this side concern from your overall proposal to move to 12s.